src/async_impl/client.rs

```diff
@@ -1340,6 +1343,16 @@ impl ClientBuilder {
             .insert(domain.to_string(), addrs.to_vec());
         self
     }
+
+    /// Override the DNS resolver implementation.
+    ///
+    /// Pass an `Arc` wrapping a trait object implementing `Resolve`.
+    /// Overrides for specific names passed to `resolve` and `resolve_to_addrs` will
+    /// still be applied on top of this resolver.
+    pub fn dns_resolver(mut self, resolver: Arc<dyn Resolve>) -> ClientBuilder {
```

Comment:
This is another question, more around ease for the user: is it annoying to convert a type into a `dyn Resolve`? I just noticed that the [`cookie_provider`}(https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.cookie_provider) method does a similar thing, but it takes an `Arc<impl CookieStore>` and then we cast it to a `Arc<dyn CookieStore>` ourselves. I no longer remember, but my guess is that the `cookie_provider` way is a little easier for users?

Reply:
Updated - that approach seems potentially more convenient.